src/coreclr/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs

```diff
@@ -2045,13 +2045,13 @@ private static void SplitName(string? fullname, out string? name, out string? ns
             int nsDelimiter = fullname.LastIndexOf('.');
             if (nsDelimiter >= 0)
             {
-                ns = fullname.Substring(0, nsDelimiter);
+                ns = fullname.AsSpan(0, nsDelimiter);
                 int nameLength = fullname.Length - ns.Length - 1;
                 if (nameLength != 0)
                     name = fullname.Substring(nsDelimiter + 1, nameLength);
                 else
                     name = "";
-                Debug.Assert(fullname.Equals(ns + "." + name));
+                Debug.Assert(fullname.Equals(ns.ToString() + "." + name));
```

Comment:
```suggestion
                Debug.Assert(fullname.Equals($"{ns}.{name}"));
```

Reply:
We should not be using instance `Equals` method of string, but `==` instead.

Reply:
Sure.